packages/lean-imt/src/lean-imt.ts

```diff
@@ -233,6 +233,52 @@ export default class LeanIMT<N = bigint> {
         this._nodes[this.depth] = [node]
     }
 
+    /**
+     * Updates N leaves all at once.
+     * It is more efficient than using the {@link LeanIMT#update} method N times because it
+     * prevents updating middle nodes several times. This would happen when updating leaves
+     * with common ancestors. However, it doesn't offer a better worst-case time complexity.
+     * @param indices The list of indices of the respective leaves.
+     * @param leaves The list of leaves to be updated.
+     */
+    public updateMany(indices: number[], leaves: N[]) {
+        requireDefined(leaves, "leaves")
+        requireDefined(indices, "indices")
+        requireArray(leaves, "leaves")
+        requireArray(indices, "indices")
+
+        if (leaves.length !== indices.length) {
+            throw new Error("There is no correspondence between indices and leaves")
+        }
+        for (let leaf = 0; leaf < indices.length; leaf += 1) {
+            requireNumber(indices[leaf], `index ${leaf}`)
+            if (indices[leaf] < 0 || indices[leaf] >= this.size) {
+                throw new Error(`Index ${leaf} is out of range`)
+            }
+        }
+
+        // This will keep track of the outdated nodes of each level.
+        let modifiedIndices = new Set<number>()
+        // First, modify the first level, which consists only of raw, un-hashed values
+        for (let leaf = 0; leaf < indices.length; leaf += 1) {
+            this._nodes[0][indices[leaf]] = leaves[leaf]
+            modifiedIndices.add(indices[leaf] >> 1)
+        }
+
+        // Now update each node of the corresponding levels
+        for (let level = 1; level <= this.depth; level += 1) {
+            const newModifiedIndices: number[] = []
+            for (const index of modifiedIndices) {
+                const leftChild = this._nodes[level - 1][2 * index]
+                const rightChild = this._nodes[level - 1][2 * index + 1]
+                this._nodes[level][index] = rightChild ? this._hash(leftChild, rightChild) : leftChild
+                newModifiedIndices.push(index >> 1)
+            }
+            modifiedIndices.clear()
```

Comment:
This variable is re-assigned in the next line, so the old set object will be replaced, and if there are no other references to it, it will be garbage collected anyway. Calling `clear` shouldn't be necessary here.

Reply:
Thanks! Forgot that line from a previous approach. Fixed.